File: backend/src/main/java/unpsjb/labprog/backend/model/enums/TipoDesignacion.java
package unpsjb.labprog.backend.model.enums;

/**
 * Enumeración que representa los tipos de designación disponibles en el
 * sistema. Distingue entre cargos administrativos y espacios curriculares de
 * enseñanza.
 *
 * @author Mariano Bonansea
 * @version 1.0
 */
public enum TipoDesignacion {
    CARGO("Cargo"),
    ESPACIO_CURRICULAR("Espacio Curricular");

    private final String valor;

    TipoDesignacion(String valor) {
        this.valor = valor;
    }

    /**
     * Obtiene la representación textual del tipo de designación.
     *
     * @return valor del tipo de designación para mostrar en la interfaz
     */
    public String getValor() {
        return valor;
    }
}

File: frontend/cli/src/app/models/tipo-designacion.ts
/**
 * Enumeración de tipos de designación docente en el sistema académico.
 * 
 * Define las categorías principales de designaciones que pueden asignarse
 * a los docentes, diferenciando entre cargos directivos/administrativos
 * y espacios curriculares específicos. Cada tipo tiene reglas de validación
 * y comportamientos particulares en el sistema de gestión.
 * 
 * @enum TipoDesignacion
 * @author Mariano Bonansea
 * @version 1.0
 */
export enum TipoDesignacion {
    /** Designación en cargo directivo o administrativo (ej: Director, Secretario) */
    CARGO = 'CARGO',

    /** Designación en espacio curricular específico (ej: Matemática, Lengua) */
    ESPACIO_CURRICULAR = 'ESPACIO_CURRICULAR',
}